src/gc/env/gcenv.ee.h

```diff
@@ -78,7 +80,7 @@ class GCToEEInterface
 
     static void GcEnumAllocContexts(enum_alloc_context_func* fn, void* param);
 
-    static void AttachCurrentThread(); // does not acquire thread store lock
+    static bool CreateBackgroundThread(Thread** thread, GCBackgroundThreadFunction threadStart, void* arg);
```

Comment:
It would not work on REDHAWK, we will return from this function even before the thread pointer is known. The value is written there by the thread itself once it starts running.